compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5674,21 +5703,54 @@ TR::Register *commonStoreEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op,
      * if valueChild is a compressed refs sequence of address constant NULL,
      * or valueChild is a zero constant integer.
      */
+    bool inZeroReg = false;
+    TR::Register *srcReg;
     if ((valueChildRoot != NULL)
         || (valueChild->getDataType().isIntegral() && valueChild->isConstZeroValue()
             && (valueChild->getRegister() == NULL))) {
-        TR::Register *zeroReg = cg->allocateRegister();
-        generateMemSrc1Instruction(cg, op, node, tempMR, zeroReg);
+        srcReg = cg->allocateRegister();
+        inZeroReg = true;
+    } else {
+        srcReg = cg->evaluate(valueChild);
+    }
+
+    if (sym->isAtLeastOrStrongerThanAcquireRelease() && tempMR->getUnresolvedSnippet() == NULL) {
+        TR_ASSERT_FATAL_WITH_NODE(node, (size == 1) || (size == 2) || (size == 4) || (size == 8),
+            "operand size for stlr must be 1, 2, 4, or 8 bytes");
```

Comment:
There is a call to `commonStoreEvaluator()` for storing a 16-byte vector.
https://github.com/eclipse-omr/omr/blob/aa214b485f6446741a1caf5855ed0d38bc6c5b89/compiler/aarch64/codegen/OMRTreeEvaluator.cpp#L5776-L5779